code/modules/events/minor/special_order.dm

```diff
@@ -0,0 +1,401 @@
+	var/list/datum/commodity/order_items
+	var/obj/storage/crate/C
+	var/obj/item/paper/requisition
+	var/list/stamps
+	var/price
+	var/weight = 100
+
+	New()
+		..()
+		var/list/datum/commodity/shopping_list = list()
+		update_requisition(requisition)
+		for(var/i in 1 to length(order_items))
+			if(ispath(order_items[i]))
+				var/path = order_items[i]
+				for(var/j in 1 to max(1,order_items[order_items[i]]))
+					shopping_list += new path
+			else
+				shopping_list += order_items[i]
+		order_items = shopping_list
+
+	proc/check_order(obj/storage/crate/sell_crate)
+		var/contents = list()
+		contents += sell_crate.contents
+		for(var/obj/item/storage/S in contents)
+			contents |= S.get_all_contents()
+
+		. = TRUE
+		for(var/datum/commodity/item as anything in order_items)
+			var/found = null
+			for(var/obj/O in contents)
+				if( ( item.subtype_valid && istype(O, item.comtype)  )  \
+				 || (!item.subtype_valid && (O.type == item.comtype) ) )
+					if(item.item_check(O))
+						found = O
+						contents -= O
+						break
+			if(!found)
+				return FALSE
+
+	proc/update_requisition(obj/item/paper/requisition)
+		if(ispath(requisition))
+			requisition = new requisition
+		requisition.info = replacetext(requisition.info, "%ITEMS%", src.get_shopping_list())
+		requisition.info += "<BR/><BR/>Requisition Offer: <B>[price]</B>"
+
+	proc/get_shopping_list()
+		. = "<ul>"
+		for(var/datum/commodity/C as anything in src.order_items)
+			if(src.order_items[C])
+				. += "<li>([src.order_items[C]]) [initial(C.comname)]</li>"
+			else
+				. += "<li>[initial(C.comname)]</li>"
+		. += "</ul>"
+
+	proc/pack_crate()
+		return
+
+ABSTRACT_TYPE(/datum/special_order/reagents)
+/datum/special_order/reagents
+	var/list/reagents_list
+
+	check_order(obj/storage/crate/sell_crate)
+		var/reagents_count = reagents_list?.Copy()
+		var/contents = list()
+		contents += sell_crate.contents
+		for(var/obj/item/storage/S in contents)
+			contents |= S.get_all_contents()
+		for(var/obj/item/reagent_containers/C in contents)
+			for(var/I in reagents_count)
+				reagents_count[I] -= C.reagents.get_reagent_amount(I)
+		for(var/I in reagents_count)
+			if(reagents_count[I] > 0)
+				return FALSE
+		return TRUE
+
+	get_shopping_list()
+		. = "<ul>"
+		for(var/R in reagents_list)
+			if(src.reagents_list[R])
+				. += "<li>([src.reagents_list[R]]u) [R]</li>"
+		. += "</ul>"
+
+ABSTRACT_TYPE(/datum/special_order/surgery)
+/datum/special_order/surgery
+	get_shopping_list()
+		. = ""
+
+/datum/commodity/special_order/pizza
+	comname = "Pizza"
+	comtype = /obj/item/reagent_containers/food/snacks/pizza
+	subtype_valid = TRUE
+
+/datum/commodity/special_order/megaweed
+	comname = "Rainbow Weed"
+	comtype = /obj/item/plant/herb/cannabis/mega
+
+/datum/commodity/special_order/whiteweed
+	comname = "White Weed"
+	comtype = /obj/item/plant/herb/cannabis/white
+
+/datum/commodity/special_order/omegaweed
+	comname = "Omega Weed"
+	comtype = /obj/item/plant/herb/cannabis/omega
```

Comment:
do these need (yet another) commodity datum? Any reason not to reuse `/datum/commodity/drugs/cannabis_mega` etc?

Reply:
Nope, artifact of proof of concept.

Reply:
Removed